### DISCUSSION TITLE
A bit confused by the `Either` usage.

### DISCUSSION THREAD

**Discussion by Magicloud** [rwf2/Rocket#2869] [Questions] [answered]
`impl<'r, 'o: 'r, 't: 'o, 'e: 'o, T, E> Responder<'r, 'o> for Either<T, E>`
Here, assuming `E` means error. Then this is similar as `Result<T, E>`.

`impl<L, R> From<Result<R, L>> for Either<L, R>`
But here, `Either` implies the first type as error (the second type in `Result`).

So this wouldn't work, if we say `E` in the first line means error.
```
fn handler() -> Either<(), Error> {
  (Ok(()) as Result<(), Error>).into()
}
```

Of course, right this does not cause issue since how to process left or right value depends on their types, not position.

**Comment by the10thWiz** [comment, marked as answer]
Rocket (and Either in general) does not treat either Left or Right as an error.

When you use Either as a responder, it simply requires both the Left and Right types to implement Responder.
